### ISSUE TITLE
"link is undefined" when accessing a resource that does not exist

### ISSUE THREAD

**Issue by ruskiyos** [LuvDaSun/angular-hal#25]
Disclaimer: this is my first attempt in getting involved in open source, so please bear with me :).

In the case that the API does not return an expected resource, when I request a resource, $get('some resource'), I get a Javascript error "link is undefined". I've traced the issue back to "angular-hal.js" line: 51. Since the resource is not provided in "_links" or "_embedded", looking it up in the "links" object will return "undefined".

I understand that if the resource isn't there, that it is not and issue with HAL (since the API should include the resource if it's there, and the client shouldn't be requesting a resource that doesn't exist), but I would expect the Hal Parser to handle that case instead of crashing.

Would this be a valid solution to add to line 138 - to reject the promise if a link doesn't exist?

```javascript
if (link === undefined) { return $q.reject('Resource not found'); }
```

Thanks.

**Comment by avandecreme** [comment]
I think you can do this in your code:

```
return $has('resource') ? $get('resource') : $q.reject('Resource not found');
```

**Comment by ruskiyos** [comment]
lol. Good point. Not sure how I skipped right over that.

But say someone (like me) didn't see that part ("$has") in the code, and keeps getting the Javascript error. Would the proposed fix be appropriate? Like I said, I'm new to open source and still not certain about the correct ways to contribute lol.

**Comment by elmerbulthuis** [comment]
this could be usefull, it would be great if someone would create a pr (including a test!)

**Comment by maennchen** [comment]
This should be fixed in PR #35.
